app/controllers/application_controller.rb

```diff
@@ -1,9 +1,14 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
 
-  helper_method :current_user
+  before_action :update_allowed_parameters, if: :devise_controller?
 
-  def current_user
-    @current_user ||= User.first
+  protected
+
+  def update_allowed_parameters
+    devise_parameter_sanitizer.permit(:sign_up) { |u| u.permit(:name, :bio, :photo, :email, :password) }
+    devise_parameter_sanitizer.permit(:account_update) do |u|
+      u.permit(:name, :bio, :photo, :email, :password, :current_password)
+    end
   end
```

Comment:
- [ ] Good job you added `allowed_parameters`, When I try to sign up I got this error `1 error prohibited this user from being saved: Name can't be blank` but there is no name input field in the signup form to fill in the first place. Please make sure that all allowed parameters have their input fields then test the sign-up and sign-in processes and ensure that everything works.

![image](https://github.com/rmiyoyo/blog-app/assets/68796232/5ff0e34c-6ec0-4f77-ac7e-52b92ac44e0e)